### PR TITLE
HIVE-27943: NPE in VectorMapJoinCommonOperator.setUpHashTable when running query with join on date

### PR DESCRIPTION
**Pull request by zabetak**
### What changes were proposed in this pull request?
Handle a `hashTableKeyType` of type `DATE` inside `setUpHashTable`

### Why are the changes needed?
Avoid the NPE by ensuring that hashTable is initialized when the key is of type `DATE`

### Does this PR introduce _any_ user-facing change?
No runtime query failures when joining on dates using vectorized map join.

### Is the change a dependency upgrade?
No

### How was this patch tested?
```
mvn test -Dtest=TestMiniLlapLocalCliDriver -Dqfile=mapjoin_date.q
```

### CODE REVIEW

**Review by aturoczy — APPROVED** [review]
+1

I know it is not necessary for the qtest, but I always prefer to do it in two separate tables. Logically the test is correct.

File: ql/src/test/queries/clientpositive/mapjoin_date.q [hunk 1]
```diff
@@ -0,0 +1,6 @@
+set hive.auto.convert.join=true;
+
+CREATE TABLE person (fname string, birthDate date);
+INSERT INTO person VALUES ('Victor', '2023-11-27'), ('Alexandre', '2023-11-28');
+
+SELECT * FROM person p1 INNER JOIN person p2 ON p1.birthDate=p2.birthDate;
```
Comment (kasakrisz) [inline ql/src/test/queries/clientpositive/mapjoin_date.q:6]: Should explain of this query added? Just to confirm that this is really a mapjoin

Comment (zabetak) [reply ql/src/test/queries/clientpositive/mapjoin_date.q:8]: Added explain in
https://github.com/apache/hive/pull/4929/commits/be930513efd36ef6ea08b281abb43025ad2059e6

**Review by kasakrisz — APPROVED** [review]
+1, pending tests

**Pull request closed without merging**